Grid.js

```diff
@@ -0,0 +1,130 @@
+                let cell = new Cell(row, col);
+                this.cells.push(cell);
+            }        
+        }
+    }
+
+    getCellByRowCol (row, col){
+        let cell = this.cells.find(c => c.row == row && c.col == col);
+        return cell;
+    }
+    
+    assignMines(){ 
+        let numMinesCreated = 0;
+        while (numMinesCreated < this.numMines) {
+            let numGridCells = this.gridSize**2;
+            let randIdx = getRandomInt(numGridCells - 1);
+            let cell = this.cells[randIdx];
+            if (!cell.isMine) {
+                cell.isMine = true;
+                numMinesCreated++;
+            }
+        }
+    };
+
+    applyFnToNeighbors(cell, fn){
+        let row = cell.row;
+        let col = cell.col;
+
+        let topLeft = {row: row-1, col: col-1};
+        let topCenter = {row: row-1, col: col};
+        let topRight = {row: row-1, col: col+1};
+        
+        let centerLeft = {row: row, col: col-1};
+        let centerRight = {row: row, col: col+1};
+    
+        let bottomLeft = {row: row+1, col: col-1};
+        let bottomCenter = {row: row+1, col: col};
+        let bottomRight = {row: row+1, col: col+1};
+    
+        let neighborCoordinates = [topLeft, topCenter, topRight, 
+                                centerLeft, centerRight, 
+                                bottomLeft, bottomCenter, bottomRight];
+       
+        const neighborCells = neighborCoordinates
+            .filter(c => this.isCellInBounds(c))
+            .map(c => this.getCellByRowCol(c.row, c.col));
+        neighborCells.forEach(neighborCell => fn(neighborCell));
+    };
+
+    isCoordinateInBounds(coordinate){
+        return coordinate >= 0 && coordinate < this.gridSize;
+    }
+
+    isCellInBounds(cell){
+        return this.isCoordinateInBounds(cell.row) && this.isCoordinateInBounds(cell.col);
+    }
+
+    isMineAndInBounds(surroundingCell) {
+        if (!this.isCellInBounds(surroundingCell)) {
+            return false;  
+        }
+        return surroundingCell.isMine;
+    }
+
+    updateThisNeighborMineCount(cell, neighborCell){
+        if(this.isMineAndInBounds(neighborCell))
+            cell.mineNeighborCount++;
+    }
+
+    updateEachCellMineNeighborCount(){
+        this.cells
+            .filter(cell => !cell.isMine)
+            .forEach(cell => {
+                cell.mineNeighborCount = 0;
+                this.applyFnToNeighbors(cell, neighborCell => this.updateThisNeighborMineCount(cell, neighborCell));
+        })
+    }
+
+    revealNeighbors(cell, cellsToReveal, zeros){
+                
+        if(!this.isCellInBounds(cell)) {
+           return;
+        }
+
+        if(cell.isRevealed){
+            return;
+        }
+
+        if (cell.mineNeighborCount == 0){ 
+            zeros.push(cell);
+        }
+        cell.isRevealed = true;
+        cellsToReveal.push(cell);
+    }
+
+    getCellsAroundZeros  = function(cell) {
```

Comment:
`Around`... do you mean something like `Neighbor`? in the naming?

Reply:
hinting at something similar to the suggestion below, like `getNeighboringZeroCells`